app/controllers/user.js

```diff
@@ -87,3 +88,38 @@ exports.userList = async (req, res, next) => {
     next(err);
   }
 };
+
+exports.singUpAdmins = async (req, res, next) => {
+  const auth = req.headers[AUTHORIZATION];
+
+  const plainText = decoder(auth);
+
+  try {
+    const requester = await User.getUserBy(plainText.email);
+
+    const permissionsErrors = validateAdmin(requester);
+
+    if (!permissionsErrors.valid) throw errors.authorizationError(permissionsErrors.messages);
+
+    const user = req.body
+      ? {
+          firstName: req.body.firstName,
+          lastName: req.body.lastName,
+          email: req.body.email,
+          password: req.body.password
+        }
+      : {};
+
+    const signErrors = validateUser(user);
+    if (!signErrors.valid) throw errors.signupError(signErrors.messages);
```

Comment:
some duplicate logic with create user in this lines, don't you think ?

Reply:
what do you do suggest for resolve that? a different name or what? I don't know what else

Reply:
Maybe a function in the same controller, doing the similar things